brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.brave;
+
+import static com.linecorp.armeria.common.brave.RequestContextCurrentTraceContext.ensureScopeUsesRequestContext;
+
+import java.util.function.Function;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.client.tracing.HttpTracingClient;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.brave.RequestContextCurrentTraceContext;
+
+import brave.Tracing;
+import brave.http.HttpTracing;
+
+/**
+ * Decorates a {@link Client} to trace outbound {@link HttpRequest}s using
+ * <a href="https://github.com/openzipkin/brave">Brave</a>.
+ *
+ * <p>This decorator puts trace data into HTTP headers. The specifications of header names and its values
```

Comment:
Think we can remove this, or otherwise make it more generic. I think the first version managed header keys in our code, but for a while we use `Tracing.injector` so the user can configure this to be anything, not necessarily the zipkin propagation.